[3단계 미션] 박예은 미션 제출합니다

ye6194 opened the pull request:
안녕하세요!
이전에도 리액트를 쓰면서 useState를 써보기도 했고, 미션 요구 사항이 음식점 모달을 띄우고 닫는 것뿐이어서 이번 3단계 미션은 쉽게 했던 것 같아요. 

## 구현 내용
![image](https://github.com/user-attachments/assets/b727c045-1726-436e-9697-aefee21c830e)
`const [isModalOpen, SetIsModalOpen] = useState(false);`: 클릭한 음식점 정보를 띄우는 모달 제어
`const [restaurantName, setRestaurantName] = useState("");`: 클릭한 음식점 상호명
`const [restaurantInfo, setRestaurantInfo] = useState("");`: 클릭한 음식점의 정보(설명)

모달이 열리는 과정: 
1. RestaurantList에서 isModalOpen = true가 되고, restaurantName과 restaurantInfo 설정됩니다.
2. isModalOpen이 true가 되면서 모달이 열리고, (`{isModalOpen && <RestaurantDetailModal ... />}`) restaurantName과 restaurantInfo이 모달에 표시됩니다.

<br/>
<br/>
<br/>

![image](https://github.com/user-attachments/assets/91a3e8dd-d676-4067-bcf6-e6561c96976c)
```html
<div className="modal-backdrop" onClick={handleClick}></div>

<button className="button button--primary text-caption" onClick={handleClick}>
    닫기
</button>
```

위 두 부분을 통해 backdrop이나 닫기 버튼을 클릭하면 모달이 닫히는 것을 구현했습니다.
지금 pr을 쓰면서 느끼는 건데 handleClick을 없애고
`<div className="modal-backdrop" onClick={() => setIsModalOpen(false)}></div>`
`<button className="button button--primary text-caption" onClick={()=>setIsModalOpen(false)}>`
이렇게 쓸걸 그랬네요. 저는 이 방식이 불필요한 함수를 쓰지 않아서 좋은 것 같은데 리뷰어님은 이 방식과 원래 방식 중 뭐가 더 낫다고 생각하시나요?? 
(그리고 사진을 보면 setIsModalOpen인데 SetIsModalOpen으로 썼네요😅 코드리뷰 후 다른 수정사항이랑 같이 고치겠습니다!)


## structure
![image](https://github.com/user-attachments/assets/dda0235b-bfac-4310-93e9-15920176ded0)

## 실행영상
https://github.com/user-attachments/assets/18106212-28cc-48b8-8d12-ffb04c8445b6

---

브랜치를 만들 때 뭐가 잘못된 것 같은데
- feat: 클릭한 레스토랑의 정보를 모달에 표시
- feat: backdrop을 클릭하면 모달이 닫힘
이 두개가 setp3입니다😢

ye6194 committed:
feat: 초기세팅

ye6194 committed:
feat: 초기세팅

ye6194 committed:
feat: 컴포넌트 그리기

ye6194 committed:
feat: 스타일 적용

ye6194 committed:
Merge branch 'step1' of https://github.com/ye6194/self-paced-react into step1

ye6194 committed:
feat: 카테고리 필터에 따라 필터된 음식점 목록을 보여줌

ye6194 committed:
feat: 필터된 음식점 목록을 보여줄 때 카테고리에 맞는 이미지 출력

ye6194 committed:
chore: 불필요한 주석 제거

ye6194 committed:
refactor: data를 다른 파일로 분리

- import 해서 map() 사용

ye6194 committed:
refactor: 코드리뷰 반영

- 불필요한 코드 삭제
- 파일 분리 (categoryInEnglish)
- export, import 방식 변경

ye6194 committed:
chore: 불필요한 설정 제거

- uninstall react-scripts

ye6194 committed:
refactor: useEffect 삭제

ye6194 committed:
fix: CRA에서 Vite로 제대로 바꿔지지 않았던 문제 해결

- 화면은 그려지지만 개발자 도구의 콘솔에 아무것도 뜨지 않고 필터가 작동하지 않는 문제가 있었음
- index.js 파일 삭제
- /public/index.html을 루트로 옮김
- 원래 루트에 있던 index.html을 templates 폴더로 옮김
- public에 있던 .css, .png 파일들을 templates 폴더로 옮김
- App.jsx의 import문에서 확장자 명시

ye6194 committed:
refactor: 코드 리뷰 반영

- categoryData, categoryInEnglish를 UPPER_SNAKE_CASE로 변경
- CATEGORY_IN_ENGLISH을 RestaurantList.jsx로 이동

ye6194 committed:
push

ye6194 committed:
Merge remote-tracking branch 'upstream/ye6194' into step2

ye6194 committed:
commit

ye6194 committed:
feat: 클릭한 레스토랑의 정보를 모달에 표시

ye6194 committed:
feat: backdrop을 클릭하면 모달이 닫힘

gogo1414 commented on src/App.jsx:
```diff
@@ -1,21 +1,47 @@
-        <CategoryFilter />
-        <RestaurantList />
+        <CategoryFilter category={category} onChangeCategory={setCategory} />
+        <RestaurantList
+          restaurants={filteredRestaurants}
+          SetIsModalOpen={SetIsModalOpen}
+          setRestaurantName={setRestaurantName}
+          setRestaurantInfo={setRestaurantInfo}
+        />
```
줄을 나누어 표현해서 보기 편하네요👍👍

ye6194 replied:
prettier를 쓰시는지 안쓰시는지는 모르지만 혹시 안쓰신다면 한번 써보세요!
잘못된 들여쓰기 수정, 코드가 길어지면 줄바꿈, 코드 끝에 세미콜론 붙이기 등 저장할 때마다 코드 정리가 돼요.
이것도 prettier가 자동으로 줄바꿈 해준 거예요.
prettier에 한번 발을 들이면 전으로는 못돌아간답니다ㅎㅎ...

gogo1414 replied:
오, 그런게 있는줄 몰랐어요!! 잘 찾아보고 도입해봐야겠네요 감사합니다😊

gogo1414 commented on src/App.jsx:
```diff
@@ -1,21 +1,47 @@
       <aside>
-        <RestaurantDetailModal />
-        <AddRestaurantModal />
+        {isModalOpen && (
+          <RestaurantDetailModal
+            SetIsModalOpen={SetIsModalOpen}
+            restaurantName={restaurantName}
+            restaurantInfo={restaurantInfo}
+          />
```
요구 사항에 맞게 구현하신 점 좋네요👍👍

gogo1414 commented on src/components/AddRestaurantModal.jsx:
```diff
@@ -1,48 +1,44 @@
+            <label htmlFor="category text-caption">카테고리</label>
+            <select name="category" id="category" required>
+              <option value="">선택해 주세요</option>
+              {CATEGORY_DATA.map((category) => (
+                <option key={category}>{category}</option>
+              ))}
```
해당 부분의 `CATEGORY_DATA` 에는 전체라는 카테고리 옵션도 같이 들어가있는데, 가게를 추가할 때 전체라는 카테고리를 선택하게 될 경우 이상하지 않을까요?? 어떻게 생각하세요??🧐

ye6194 replied:
네... 사실 step3를 구현하면서 전체를 선택 후 추가해보니 제대로 작동이 안되더라구요... 인지만 하고 고치지는 않았는데 이번에 고치겠습니다

gogo1414 commented on src/components/AddRestaurantModal.jsx:
현재 HTML의 각 태그에 어떤 내용이 들어가있는지 주석으로 설명을 달고 계시네요!
HTML 태그 안에 있는 각 변수명, Props명 등이 목적과 동작을 명확하게 드러내 설명이 된다고 하면 주석을 사용하지 않아도 괜찮지 않을까요??
어떻게 생각하시는지 궁금하네요!

관련된 키워드로 [Self-Documenting Code](https://dev.to/mattlewandowski93/writing-self-documenting-code-4lga)가 있습니다!

ye6194 replied:
엇 이건 제가 의도하고 적은 주석이 아니고 [step1](https://github.com/cho-log/self-paced-react/tree/main/01-first-component) 미션 때 '/templates 에 있는 html 템플릿을 그대로 App.jsx에서 그릴 수 있게 해보세요.'라는 요구사항을 충족시키면서 생긴 주석입니다.

![image](https://github.com/user-attachments/assets/c455a417-46cf-4e85-8612-b2bd6ff44985)
예를 들어서 templates/index.html의 코드는 이렇게 되어있는데 이 코드를 그대로 복사해서 붙여넣는 과정에서 주석이 있으면 무슨 코드인지 알아보기 쉬울 것 같아서 굳이 빼지 않았어요.

링크의 글은 잘 읽어보았습니다. 그래도 저는 이 정도의 주석은 가독성을 해치는 정도도 아니고, 아무리 코드를 잘 작성했어도 영어보다는 한국어가 더 눈에 잘띄기도 해서(한국인한정..) 주석이 있는 편이 더 마음에 드네요.
또 나중에 '추가 버튼'이 어딨는지 찾고 싶다고 하면 ctrl+F로 찾기도 편하구요.

하지만 명확하게 작성된 코드에 굳이 주석을 달지 않는게 클린 코드라면 따르는게 맞겠죠🥹
덕분에 self-documenting code라는 개념을 새롭게 알고 갑니다!

gogo1414 commented on src/components/RestaurantDetailModal.jsx:
```diff
@@ -1,18 +1,24 @@
 
-function RestaurantDetailModal() {
+function RestaurantDetailModal({ SetIsModalOpen, restaurantName, restaurantInfo }) {
+  const handleClick = () => {
+    SetIsModalOpen(false);
+  };
```
예은님이 말씀하신 것처럼 `handleClick`이 존재하지 않아도 괜찮을거 같아요! 다만 저렇게 사용했을 때 이점이 존재하면 저런 방법도 고려해봐도 좋지 않을까요? 그런생각이 드네요!

저는 아래와 같이 작성해 봤습니다!
```javascript
function RestaurantDetailModal({ restaurant, onChangeModal }) {
    return (
        <div className={`${styles.modal} ${styles["modal--open"]}`}>
            <div className={styles["modal-backdrop"]} onClick={onChangeModal}></div>
            <div className={styles["modal-container"]}>
                <h2 className={`${styles["modal-title"]} text-title`}>{restaurant.name}</h2>
                <div className={styles["restaurant-info"]}>
                    <p className={`${styles["restaurant-info__description"]} text-body`}>{restaurant.description}</p>
                </div>
                <div className={styles["button-container"]} onClick={onChangeModal}>
                    <button className={`${styles.button} ${styles["button--primary"]} text-caption`}>닫기</button>
                </div>
            </div>
        </div>
    );
}
```

ye6194 replied:
준수님 코드도 좋은 것 같습니다👍

gogo1414 reviewed (changes requested):
안녕하세요, 예은님!
리뷰어 김준수입니다😀

3단계 미션은 쉽게 하셨다니 다행이네요!
모달이 열리는 과정을 글로 표현해주셔서 이해하기가 쉬웠네요!

마지막에 작성해주신 것처럼 마지막 2개의 커밋만 필터를 걸어서 봤습니다! (까먹고 쭉 보다가 중간에 기억났어요!) 해당 커밋이 아닌 다른 곳에도 리뷰를 달았는데 고민해볼만한 내용 같아서 참고하셔서 보시면 좋을거 같네요!

> 지금 pr을 쓰면서 느끼는 건데 handleClick을 없애고 <div className="modal-backdrop" onClick={() => setIsModalOpen(false)}></div> <button className="button button--primary text-caption" onClick={()=>setIsModalOpen(false)}>이렇게 쓸걸 그랬네요. 저는 이 방식이 불필요한 함수를 쓰지 않아서 좋은 것 같은데 리뷰어님은 이 방식과 원래 방식 중 뭐가 더 낫다고 생각하시나요??
해당 부분에 대해서는 리뷰로 달았습니다! 확인해보시면 좋을거 같네요😊

3단계 미션도 고생하셨습니다!

ye6194 commented:
안녕하세요 준수님! 

이번 코드리뷰를 통해 self-documenting code(자기 문서화 코드)를 알게되었습니다. 감사합니다😊

그리고
> 마지막 2개의 커밋만 필터를 걸어서 봤습니다! 

이걸 보고 저도 필터거는걸 찾아보고 써봤는데 되게 편리하네요!
이번에 새로운 것을 두개나 알게 되어서 좋아요👍

ye6194 committed:
refactor: 코드리뷰 내용 반영 및 수정

- 불필요한 주석 삭제
- RestaurantDetailModal에서 handleClick 함수 삭제
- CategoryFilter에서 handleChange 함수 삭제
- 새로운 음식점 추가 시 카테고리에서 '전체' 항목 삭제

ye6194 committed:
refactor: 함수 구조 변경 및 오타 수정

gogo1414 reviewed (approved):
리팩터링 하시느라 고생하셨습니다!!

Indigochi1d commented on .eslintrc.cjs:
```diff
@@ -15,5 +15,6 @@ module.exports = {
     "react/jsx-no-target-blank": "off",
     "react/prop-types": "off",
     "react-refresh/only-export-components": ["warn", { allowConstantExport: true }],
+    "no-unused-vars": "warn",
```
eslint에서 사용하지 않는 변수를 찾아내는 설정을 해놓고 만드셨군요!! 좋습니다👍🏻

ye6194 replied:
사실 이건 제가 추가한 기억이 없어서 왜 있는지 모르겠는데...!! 사용하지 않는 변수에 경고를 띄우는 설정이었네요.

Indigochi1d commented on src/App.jsx:
```diff
@@ -1,21 +1,47 @@
+
+  const filteredRestaurants = filterRestaurants(category);
+
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const [restaurantName, setRestaurantName] = useState("");
+  const [restaurantInfo, setRestaurantInfo] = useState("");
```
모달을 구현하기 위해서 useState를 여러개 사용하셨군요! 좋습니다. 하지만 만약 모달에 띄워야하는 정보가 5~6개 가량 된다고 한다면  상태선언 Hook을 많이 쓰는건 코드 가독성과 성능에 있어 비효율적일 것 같은데 예은님은 어떻게 생각하시나요,,?
그래서 제가 제안드리는 방법은 아래의 코드와 같습니다.
```javascript
const [isModal, setIsModal] = useState({
        isOpen: false,
        restaurants:{
            name:"",
            description:"",
        }
 });
```
이렇게 아예 객체로 선언해서 상태를 담는다면 코드 유지보수성과 의미성, 성능에 더 도움이 될 것 같다고 개인적인 생각을 가지고 있어요. 예은님의 생각이 어떤지 궁금하네요..!

ye6194 replied:
오 맞아요! 다른 프로젝트할 때 한 컴포넌트에서 고객의 이름, 예약한 곳, 원하는 날짜, 원하는 시간, 원하는 방법 등 띄워야 하는 정보가 많아서 useState가 엄청 많았던 기억이 있어요.
저는 useState의 초기값에 정수, 불리언 값, 빈 문자열만 넣어봐서 객체를 넣는다는건 생각도 못했네요. 객체를 넣으면 범수님 말씀대로 useState를 쓸데없이 많이 쓰지 않아도 돼서 코드 가독성과 성능 향상에 도움이 될 것 같아요!

Indigochi1d commented on src/App.jsx:
```diff
@@ -1,21 +1,47 @@
+        <CategoryFilter category={category} onChangeCategory={setCategory} />
+        <RestaurantList
+          restaurants={filteredRestaurants}
+          setIsModalOpen={setIsModalOpen}
+          setRestaurantName={setRestaurantName}
+          setRestaurantInfo={setRestaurantInfo}
```
이곳에서도 아까와 마찬가지로 객체로 선언했다면 `setIsModal` 이라는 setter함수만 넘겨주면 되기에 코드 가독성에 더 도움이 될 수 있을 것 같아요👀

ye6194 replied:
여기도 객체로 고쳐보겠습니다🙂

Indigochi1d commented on src/components/AddRestaurantModal.jsx:
```diff
@@ -1,48 +1,40 @@
+        <form>
+          <div className="form-item form-item--required">
+            <label htmlFor="category text-caption">카테고리</label>
+            <select name="category" id="category" required>
+              <option value="">선택해 주세요</option>
+              {CATEGORY_DATA.slice(1).map((category) => (
+                <option key={category}>{category}</option>
+              ))}
+            </select>
```
CATEGORY_DATA를 slice해서 `전체`를 걸러내셨군요! 4단계 미션에서 구현해야하는 내용인데 미리 구현하셨네요👏🏻

Indigochi1d commented on src/components/RestaurantDetailModal.jsx:
```diff
@@ -1,18 +1,22 @@
     <>
       <div className="modal modal--open">
-        <div className="modal-backdrop"></div>
+        <div className="modal-backdrop" onClick={() => setIsModalOpen(false)}></div>
```
구현을 꼼꼼히 잘 하셨네요👍🏻 근데 이 때 onClick 내부에서
```javascript
onClick = { setIsModalOpen() }
``` 
대신 익명(화살표)함수를 쓰신 이유를 알 수 있을까요?

ye6194 replied:
엇 그렇게 하면 backdrop을 클릭할 때 setIsModalOpen이 실행되는게 아니라 렌더링될 때 바로 setIsModalOpen이 실행돼서 제대로 작동이 안되는거 아닌가요??

Indigochi1d replied:
맞습니다! 잘 알고 계셨네요! 자세하게 설명드리면 아래의 내용과 같아요.
### 1. JavaScript의 함수 호출 방식

JavaScript에서 함수 이름 뒤에 괄호 `()`를 붙이면 그 함수는 즉시 호출됩니다. 예를 들어, `onCardClick()`처럼 작성하면 JavaScript 엔진은 이를 즉시 실행하고 그 **결과값**을 반환합니다. 

```jsx
const result = onCardClick();  
// onCardClick 함수가 즉시 실행됨
```

`onCardClick`를 **참조**하는 대신 `onCardClick()`로 **호출**하게 되면 즉시 실행이 일어나는 이유가 여기에 있어요.

### 2. JSX의 평가 순서

React 컴포넌트가 렌더링될 때, JSX 내부의 표현식(중괄호 `{ }`로 감싸진 부분)은 **모두 평가**됩니다. 즉, JSX 내에서 `{onCardClick()}`와 같은 표현식이 포함되면 렌더링 시점에 JavaScript는 이 표현식을 **즉시 실행**하고, 결과값을 JSX에 할당하게 됩니다.

```jsx
onClick={onCardClick()} 
 // 렌더링 시점에 onCardClick() 실행 -> 결과값 할당
```

React의 이벤트 핸들러 `onClick`에 할당할 때도 마찬가지로, `onClick={onCardClick()}`라고 작성하면 **렌더링 중에** `onCardClick()`이 실행됩니다. 클릭 이벤트가 발생하지 않아도 렌더링 시점에서 즉시 호출되는 이유입니다!

Indigochi1d commented on src/components/RestaurantList.jsx:
```diff
@@ -1,84 +1,40 @@
 import "../styles/RestaurantListStyle.css";
 
-function RestaurantList() {
+const CATEGORY_IN_ENGLISH = Object.freeze({
+  한식: "korean",
+  중식: "chinese",
+  일식: "japanese",
+  양식: "western",
+  아시안: "asian",
+  기타: "etc",
+});
```
👍🏻 변하지 않는 상수를 Object.freeze를 통해 고정값으로 두고 대문자로 가독성까지 챙기셨네요!!

Indigochi1d commented on src/components/RestaurantList.jsx:
```diff
@@ -1,84 +1,40 @@
+});
+
+function RestaurantList({ restaurants, setIsModalOpen, setRestaurantName, setRestaurantInfo }) {
+  const handleClick = (restaurant) => {
+    setIsModalOpen(true);
+    setRestaurantName(restaurant.name);
+    setRestaurantInfo(restaurant.description);
+  };
```
List중 아이템(점포)이 클릭이 되었을때 따로 로직을 처리하셨군요!
다만 handleClick이란 네이밍이 잘못된 것은 아니지만 `handleRestaurantClick`정도로 한다면 함수의 의미와 기능이 완전히 일치할 것 같아요!

ye6194 replied:
네 그렇게 고치면 함수의 역할이 더 명확하게 드러나겠네요!

Indigochi1d commented on src/data/categoryData.js:
```diff
@@ -0,0 +1,9 @@
+export const CATEGORY_DATA = Object.freeze([
```
freeze 👍🏻

Indigochi1d reviewed (changes requested):
안녕하세요 예은님!  잘 지내셨나요😆 
모달의 on/off 상태를 boolean으로 초기화하고 setter함수를 props로 전달해서 상태 값을 특정 컴포넌트에서 바꾸셨군요! 리액트가 설계될때부터 데이터의 단방향 흐름을 목적에 두고 했기에 그랬다는 것을 미리 알고 계셨던 예은님께서 더욱 쉽게 느끼셨을 것 같습니다! 어려운 로직이 아니지만 기본에 충실한 미션요구사항과 그것을 지켜낸 예은님의 코드라고 생각해요👍🏻 

> 지금 pr을 쓰면서 느끼는 건데 handleClick을 없애고
> ```javascript
> <div className="modal-backdrop" onClick={() => setIsModalOpen(false)}></div>
> <button className="button button--primary text-caption" onClick={()=>setIsModalOpen(false)}>
> ```
> 이렇게 쓸걸 그랬네요. 저는 이 방식이 불필요한 함수를 쓰지 않아서 좋은 것 같은데 리뷰어님은 이 방식과 원래 방식 중 뭐가 더 낫다고 생각하시나요??

라고 하셨던 것에서 저는 새로 적용한 방식이 더 좋아보입니다! 굳이 `handleClick()` 함수를 한번 더 만들 필요는 없다고 생각해요! 하지만 하나 저도 궁금한 점이 생겼는데 setter함수를 props로 넘길 때

> 코드 컨벤션이 있는지? 굳이 on 또는 handle로 네이밍을 달리해서 넘겨야하는지?

정도를 내일 토론 해보면 좋을 것 같아요. 
[리액트 공식문서](https://react.dev/learn/thinking-in-react)에서는 on 또는 handle로 네이밍해서 props로 넘기고 있지만요!
![image](https://github.com/user-attachments/assets/6c56c4a4-08e0-4b3e-87c4-4d3288dcd873)

이번주도 수고하셨습니다!!😃

ye6194 commented:
안녕하세요 범수님🙌

> 코드 컨벤션이 있는지? 굳이 on 또는 handle로 네이밍을 달리해서 넘겨야하는지?

이 부분에 대한 제 의견을 말해보자면, on이나 handle로 넘겨야 한다는 규칙이 있다면 그걸 따르는게 맞겠지만 그런 규칙이 없다면 저는 setter 함수 그대로 넘기는게 편하다고 생각합니다.
예를 들어서 `setIsModalOpen`을 props로 넘길 때 똑같이 `setIsModalOpen`으로 넘기면
![image](https://github.com/user-attachments/assets/713f3591-d9dd-43b2-9e59-01a39dbfe78f)
이렇게 버튼을 클릭했을 때 무슨 함수가 실행되는지 바로 알 수 있어서 좋은 것 같아요.

반면에 
![image](https://github.com/user-attachments/assets/7797743d-ca26-41a2-8843-b6b2c1947520)
`onDetailModalClick`으로 넘기면 '닫기 버튼을 눌렀을 때 `onDetailModalClick`이 실행되는건 알겠는데 `onDetailModalClick`이 뭐지?' 하는 생각이 들 수도 있을 것 같아요. 그럼 또 App으로 가서 `onDetailModalClick`이 무슨 함수인지 찾아봐야하는 불편함이 있구요.

하지만 자바스크립트에서 이벤트 핸들러를 지정할 때 **on**Click, **on**Chage 등의 속성을 사용하는걸 생각하면, **on**DetailModalClick을 쓰는게 통일성 측면에서 좋아보입니다. 또한 리액트 공식문서에도 on이나 handle을 붙여 넘기고 있는걸 보면 그렇게 쓰는게 일반적인 것 같기도 하네요.

<br />

요약)
- setter 함수를 그대로 넘기거나 on, handle 등으로 네이밍해서 넘기는 방법 모두 장단점이 있다.
- on, handle 등으로 네이밍해서 넘겨야 한다는 규칙이 있으면 따르는게 맞다.

Indigochi1d commented:
저와 생각이 일치하네요! 저도 setter함수의 네이밍을 그대로 두고 props로 넘겨왔어요. 왜냐하면 상태관리 라이브러리를 따로 쓰지 않는다면 상태와 상태변경 로직을 props로 넘겨가며 전달해야하는데 그 때 `on~~` 으로 바꾸면 실사용 하위컴포넌트에서 `이게 뭐였지?` 라고 생각되더라구요! 저도 얼마 전 리액트 공식문서를 보고나서  `on~~` 으로 넘기는게 일반적일까? 라고 고민을 가지게 되었습니다!

이번 주도 수고하셨고 코드 수정이 완료되시면 머지요청 멘션 디스코드로 남겨주세요! 감사합니다😸

ye6194 committed:
refactor: 코드리뷰 내용 반영

- 함수 이름 변경 (handleClick -> handleRestaurantClick)
- isModalOpen, restaurantName, restaurantInfo를 modal로 통합

ye6194 committed:
chore: 불필요한 console.log 삭제

ye6194 committed:
Merge branch 'ye6194' into step3
